### PR TITLE
Expose the ability to use default encryption keys in fscloud submodule + DA

### PR DESCRIPTION
**Pull request by ocofaigh**
### Description

- Expose the ability to use default encryption keys in fscloud submodule + DA
- Exposed the ability to set the IBM provider visibility (defaults to private)

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by ocofaigh** [comment]
Need to skip the upgrade test due to new variable being added, and the fact that it has to be set to public for upgrade test to work in GHA runtime..
```
│ Error: Value for undeclared variable
│ 
│ A variable named "provider_visibility" was assigned on the command line,
│ but the root module does not declare a variable of that name. To use this
│ value, add a "variable" block to the configuration.
╵}
```

File: cra-config.yaml [hunk 1]
```diff
@@ -9,4 +9,5 @@ CRA_TARGETS:
       TF_VAR_kms_key_crn: "crn:v1:bluemix:public:hs-crypto:us-south:a/abac0df06b644a9cabc6e44f55b3880e:e6dce284-e80f-46e1-a3c1-830f7adff7a9:key:76170fae-4e0c-48c3-8ebe-326059ebb533"
       TF_VAR_prefix: "test-postgres-standard"
       TF_VAR_use_existing_resource_group: true
-      TF_VAR_resource_group_name: "geretain-test-postgres"
+      TF_VAR_resource_group_name: "Default"
```
Comment (rajatagarwal-ibm) [inline cra-config.yaml:12]: What is the requirement to update the resource group to "Default" here?

File: modules/fscloud/README.md [hunk 2]
```diff
@@ -34,11 +34,11 @@ No resources.
...
-| <a name="input_backup_encryption_key_crn"></a> [backup\_encryption\_key\_crn](#input\_backup\_encryption\_key\_crn) | The CRN of a Hyper Protect Crypto Services use for encrypting the disk that holds deployment backups. Only used if var.kms\_encryption\_enabled is set to true. There are limitation per region on the Hyper Protect Crypto Services and region for those services. See https://cloud.ibm.com/docs/cloud-databases?topic=cloud-databases-hpcs#use-hpcs-backups | `string` | `null` | no |
+| <a name="input_backup_encryption_key_crn"></a> [backup\_encryption\_key\_crn](#input\_backup\_encryption\_key\_crn) | The Hyper Protect Crypto Services (HPCS) or Key Protect root key CRN to use for encrypting the disk that holds deployment backups. There are region limitation for backup encryption. See https://cloud.ibm.com/docs/cloud-databases?topic=cloud-databases-hpcs#use-hpcs-backups (HPCS) and https://cloud.ibm.com/docs/cloud-databases?topic=cloud-databases-key-protect&interface=ui#key-byok (Key Protect). | `string` | `null` | no |
```
Comment (rajatagarwal-ibm) [inline modules/fscloud/README.md:37]: For consistency, could you please also update the variable description at the root level?

File: solutions/standard/main.tf [hunk 3]
```diff
@@ -9,9 +9,9 @@ locals {
   key_name      = var.prefix != null ? "${var.prefix}-${var.key_name}" : var.key_name
   key_ring_name = var.prefix != null ? "${var.prefix}-${var.key_ring_name}" : var.key_ring_name
 
-  kms_key_crn = var.existing_kms_key_crn != null ? var.existing_kms_key_crn : module.kms[0].keys[format("%s.%s", local.key_ring_name, local.key_name)].crn
+  kms_key_crn = !var.use_ibm_owned_encryption_key ? var.existing_kms_key_crn != null ? var.existing_kms_key_crn : module.kms[0].keys[format("%s.%s", local.key_ring_name, local.key_name)].crn : null
 
-  create_cross_account_auth_policy = !var.skip_iam_authorization_policy && var.ibmcloud_kms_api_key != null
+  create_cross_account_auth_policy = !var.skip_iam_authorization_policy && var.ibmcloud_kms_api_key != null && !var.use_ibm_owned_encryption_key
```
Comment (rajatagarwal-ibm) [inline solutions/standard/main.tf:14]: Is there a limitation that IBM owned encryption keys cannot be used to create cross account auth policy?

**Review by rajatagarwal-ibm — CHANGES_REQUESTED** [review]

File: cra-config.yaml (hunk 1 above)
Comment (ocofaigh) [reply cra-config.yaml:12]: Default resource group cannot be deleted, but the `geretain-test-postgres` can be - so just protecting it from breaking in the future

File: solutions/standard/main.tf (hunk 3 above)
Comment (ocofaigh) [reply solutions/standard/main.tf:14]: If your using an IBM owned key, it means your not using BYOK or KYOK encryption, so no auth policy is needed

File: cra-config.yaml (hunk 1 above)
Comment (ocofaigh) [reply cra-config.yaml:12]: I think Ill just set `use_existing_resource_group` to false instead so the plan will try to create one

**Review by rajatagarwal-ibm — APPROVED** [review]

File: modules/fscloud/README.md (hunk 2 above)
Comment (ocofaigh) [reply modules/fscloud/README.md]: done

**Pull request merged**